pkg/machines/components/vcpuModal.jsx

```
@@ -0,0 +1,192 @@
+import InfoRecord from './infoRecord.jsx';
+import { setVCPUSettings } from "../actions.es6";
+
+React;
+
+const _ = cockpit.gettext;
+
+const dividers = (num) => {
+    const divs = [1];
+
+    for (let i = 2; i < num; i++) {
+        if (num % i === 0) {
+            divs.push(i);
+        }
+    }
+
+    if (num > 1) {
+        divs.push(num);
+    }
+
+    return divs;
+}
+
+const Select = function ({ id, items, onChange, value }) {
+    return ( <SelectComponent.Select id={id} initial={value} onChange={onChange}>
+        {items.map((t) => (
+            <SelectComponent.SelectEntry data={t}>{t}</SelectComponent.SelectEntry>
+        ))}
+    </SelectComponent.Select>);
+}
+
+class VCPUModalBody extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            max: props.vcpus.max || 1,
+            count: parseInt(props.vcpus.count) || 1,
+            sockets: props.cpu.topology.sockets || 1,
+            threads: props.cpu.topology.threads || 1,
+            cores: props.cpu.topology.cores || 1
+        };
+        this.onMaxChange = this.onMaxChange.bind(this);
+        this.onCountSelect = this.onCountSelect.bind(this);
+        this.onSocketChange = this.onSocketChange.bind(this);
+        this.onThreadsChange = this.onThreadsChange.bind(this);
+        this.onCoresChange = this.onCoresChange.bind(this);
+        props.onChange(this.state);
+    }
+
+    componentWillUpdate (nextProps, nextState) {
+        this.props.onChange(nextState);
+    }
+
+    onMaxChange (e) {
+        const maxHypervisor = parseInt(this.props.hypervisorMax);
+        let maxValue = parseInt(e.target.value);
+        if (isNaN(maxValue) || maxValue <= 0) {
+            maxValue = 1;
+        }
+        if (maxValue > maxHypervisor) {
+            maxValue = maxHypervisor;
+        }
+        if (maxValue < this.state.count) {
+            this.setState({ count: maxValue });
+        }
```

Does htis mean this can only ever be reduced in the UI, never increased any more? This looks like it would just silently ignore increasing the value in the dialog?

---

No, `Count` of currently used VCPU can't be greater then Max limit of VCPU. And those lines are for limitation.

---

Of course, I misread this. Thanks!